independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/CurrentManagedContext.java

```diff
@@ -62,10 +62,17 @@ public ContextState activate(ContextState initialState) {
         if (initialState == null) {
             CurrentContextState state = initializeState();
             currentContext.set(state);
+            if (state.shouldFireInitializedEvent()) {
```

Comment:
Hm, I think that we should just move `fireIfNotNull(initializedNotifier)` out of `initializeState()` and place after `currentContext.set(state)`, and move `fireIfNotNull(beforeDestroyedNotifier);` before `currentState.invalidate()`, i.e. no need to add more state to the `CurrentContextState`...

Reply:
Hm, that's what I did initially (and it works for just ArC) but then I thought we are passing around context state and we cannot be sure if the state was initialized/destroyed or not or we just set its context?

Reply:
That's a very good point.

Reply:
I think that we should also clarify that `ManagedContext#initializeState()` does not fire an `@Initialized` event. That it merely creates a new context state.

Reply:
Yes, and while at it, I will also add that `active()` fires it.